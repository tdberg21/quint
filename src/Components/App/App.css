* {
  font-family: 'Chakra Petch', sans-serif;
  font-size: 19px;
}

.quint-logo {
  text-align: left;
}

.header-title {
  font-size: 1.6rem;
  font-weight: 700;
  margin-bottom: 0;
}

.header-subtitle {
  margin-top: 0;
  font-size: 0.7rem;
  font-style: italic;
}

.App {
  text-align: center;
  border: 2rem solid #002868;
}

.App-header {
  background-color: #000;
  min-height: 10vh;
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: #fff;
  padding: 0 15px;
}

.menu-button {
  background-color: #000;
  color: #fff;
  font-size: 1.2rem;
  text-decoration: none;
  border: none;
}

.responsive-header-bar {
  margin: 0 auto;
  padding: 0px;
  display: flex;
  flex-direction: column;
  background-color: #000;
  width: 21%;
  position: absolute;
  right: 2rem;
}

.responsive-header-bar-nav-links {
  padding: 15px 0;
  color: #fff;
  text-decoration: none;
  font-weight: 300;
}

.responsive-header-bar-nav-links:hover {
  background-color: #fff;
  color: #000;
  border: 1px solid black;
}

.app-body {
  min-height: 75vh;
  background: linear-gradient(
      rgba(1, 23, 5, 0.7), 
      rgba(1, 23, 5, 0.7)), url('../../assets/moscow.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  padding-top: 3rem;
}

.blur-bg {
  width: 70%;
  margin: 0 auto 0 auto;
  min-height: fit-content;
  background: rgba(1, 23, 5, 0.75);
  padding: 1rem;
}

.home-paragraph {
  color: #fff;
  margin: 1rem auto 0 auto;
  text-align: left;
  padding-bottom: 2rem;
}

.menu-button {
  font-weight: 500;
}

@media screen and (max-width: 650px) {

  .header-title {
    font-size: 1.4rem;
  }

  .header-subtitle {
    display: none;
  }

  .menu-button {
  font-size: 0.8rem;
}

  .responsive-header-bar {
  width: 31%;
}

.responsive-header-bar-nav-links {
  font-size: 0.8rem;
}
}