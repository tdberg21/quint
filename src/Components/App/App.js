import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import { bioData } from '../../bioData.js';
import './App.css';
import Threat from '../Threat/Threat.js';
import Home from '../Home/Home.js';
import GroupInfo from '../GroupInfo/GroupInfo.js';
import ShortBioContainer from '../ShortBioContainer/ShortBioContainer.js';
import Products from '../Products/Products.js';
import LongBio from '../LongBio/LongBio.js';
import Contact from '../Contact/Contact.js';
import Testimonials from '../Testimonials/Testimonials.js';
import Publications from '../Publications/Publications.js';

class App extends Component {
  constructor() {
    super();

    this.state = {
      isHidden: true,
      pageName: ''
    }
  }

  hideMenu = () => {
    this.setState({
      isHidden: true
    })
  }

  showMenu = () => {
    this.setState({
      isHidden: false
    })
  }

  toggleHidden = () => {
    this.setState({
      isHidden: !this.state.isHidden
    })
  }

  

  render() {
    return (
      <div className='App'>
        <header className='App-header'>
          <div className='quint-logo'>
          <h1 className='header-title'>Quint Advisors</h1>
          <p className='header-subtitle'>A geopolitical, security and cyber risk advisory practice.</p>
          </div>
          <h2> </h2>
          <div className='header-menu'>
            <button 
              className='menu-button'
              onClick={() => this.toggleHidden()}
              onMouseEnter={() => this.showMenu()}
            >Menu</button>  
          </div> 
        </header>
        {!this.state.isHidden && <Menu toggleHidden={() => this.toggleHidden()} hideMenu={() => this.hideMenu()}/>}
        
        <Route exact path='/' component={Home} />
        <Route exact path='/threat' component={Threat} />
        <Route exact path='/whoweare' component={GroupInfo} />
        <Route exact path='/bios' component={ShortBioContainer} />
        <Route exact path='/products' component={Products} />
        <Route exact path='/contact' component={Contact} />
        <Route exact path='/testimonials' component={Testimonials} />
        <Route exact path='/publications' component={Publications} />
        <Route path='/bios/:name' render={({ match }) => {
          let info = bioData.find(bio => {
            return bio.lastName === match.params.name; 
          });
          return <LongBio {...info} />;
        }} />
      </div>
    );
  }
}

const Menu = (props) => (      
  <span className="display-toggle" onClick={() => props.toggleHidden()} onMouseLeave={() => props.hideMenu()}>
    <div className="responsive-header-bar">
      <NavLink to='/' className="responsive-header-bar-nav-links" value='home' onClick={(event) => setPageName(event)}>Home</NavLink>
      <NavLink to='/threat' className="responsive-header-bar-nav-links"> <button value='threat' onClick={(event) => setPageName(event)}>The Threat</button></NavLink>
      <NavLink to='/whoweare' className="responsive-header-bar-nav-links" value='home' onClick={(event) => setPageName(event)}>Who We Are</NavLink>
      <NavLink to='/products' className="responsive-header-bar-nav-links">What We Offer</NavLink>
      <NavLink to='/bios' className="responsive-header-bar-nav-links">Individual Bios</NavLink>
      <NavLink to='/contact' className="responsive-header-bar-nav-links">Contact Us</NavLink>
      <NavLink to='/testimonials' className="responsive-header-bar-nav-links">Client Testimonials</NavLink>
      <NavLink to='/publications' className="responsive-header-bar-nav-links">Publications</NavLink>
    </div>
  </span>
)

const setPageName = (event) => {
  console.log(event)
}

export default App;
