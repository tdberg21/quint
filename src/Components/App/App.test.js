import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});