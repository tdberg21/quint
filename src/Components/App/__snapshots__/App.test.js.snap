// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <div
            className="quint-logo"
          >
            <h1
              className="header-title"
            >
              Quint Advisors
            </h1>
            <p
              className="header-subtitle"
            >
              A geopolitical, security and cyber risk advisory practice.
            </p>
          </div>
          <h2>
             
          </h2>
          <div
            className="header-menu"
          >
            <button
              className="menu-button"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Menu
            </button>
          </div>
        </header>,
        false,
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/threat"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/whoweare"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/bios"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/products"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/contact"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/testimonials"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/publications"
        />,
        <Route
          path="/bios/:name"
          render={[Function]}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="quint-logo"
            >
              <h1
                className="header-title"
              >
                Quint Advisors
              </h1>
              <p
                className="header-subtitle"
              >
                A geopolitical, security and cyber risk advisory practice.
              </p>
            </div>,
            <h2>
               
            </h2>,
            <div
              className="header-menu"
            >
              <button
                className="menu-button"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Menu
              </button>
            </div>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="header-title"
                >
                  Quint Advisors
                </h1>,
                <p
                  className="header-subtitle"
                >
                  A geopolitical, security and cyber risk advisory practice.
                </p>,
              ],
              "className": "quint-logo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Quint Advisors",
                  "className": "header-title",
                },
                "ref": null,
                "rendered": "Quint Advisors",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A geopolitical, security and cyber risk advisory practice.",
                  "className": "header-subtitle",
                },
                "ref": null,
                "rendered": "A geopolitical, security and cyber risk advisory practice.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " ",
            },
            "ref": null,
            "rendered": " ",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="menu-button"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Menu
              </button>,
              "className": "header-menu",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Menu",
                "className": "menu-button",
                "onClick": [Function],
                "onMouseEnter": [Function],
              },
              "ref": null,
              "rendered": "Menu",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "header",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/threat",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/whoweare",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/bios",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/products",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/contact",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/testimonials",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/publications",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/bios/:name",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <div
              className="quint-logo"
            >
              <h1
                className="header-title"
              >
                Quint Advisors
              </h1>
              <p
                className="header-subtitle"
              >
                A geopolitical, security and cyber risk advisory practice.
              </p>
            </div>
            <h2>
               
            </h2>
            <div
              className="header-menu"
            >
              <button
                className="menu-button"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Menu
              </button>
            </div>
          </header>,
          false,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/threat"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/whoweare"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/bios"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/products"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/contact"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/testimonials"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/publications"
          />,
          <Route
            path="/bios/:name"
            render={[Function]}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="quint-logo"
              >
                <h1
                  className="header-title"
                >
                  Quint Advisors
                </h1>
                <p
                  className="header-subtitle"
                >
                  A geopolitical, security and cyber risk advisory practice.
                </p>
              </div>,
              <h2>
                 
              </h2>,
              <div
                className="header-menu"
              >
                <button
                  className="menu-button"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Menu
                </button>
              </div>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="header-title"
                  >
                    Quint Advisors
                  </h1>,
                  <p
                    className="header-subtitle"
                  >
                    A geopolitical, security and cyber risk advisory practice.
                  </p>,
                ],
                "className": "quint-logo",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Quint Advisors",
                    "className": "header-title",
                  },
                  "ref": null,
                  "rendered": "Quint Advisors",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A geopolitical, security and cyber risk advisory practice.",
                    "className": "header-subtitle",
                  },
                  "ref": null,
                  "rendered": "A geopolitical, security and cyber risk advisory practice.",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " ",
              },
              "ref": null,
              "rendered": " ",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="menu-button"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Menu
                </button>,
                "className": "header-menu",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Menu",
                  "className": "menu-button",
                  "onClick": [Function],
                  "onMouseEnter": [Function],
                },
                "ref": null,
                "rendered": "Menu",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "header",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/threat",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/whoweare",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/bios",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/products",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/contact",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/testimonials",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/publications",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/bios/:name",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
